app/core/auth.coffee

```diff
@@ -8,14 +8,16 @@ init = ->
   module.exports.me = window.me = new User(window.userObject) # inserted into main.html
   module.exports.me.onLoaded()
   trackFirstArrival()
+  # set country and geo fields for returning users if not set during account creation (/server/models/User - makeNew)
+  if not me.get('country')
+    $.ajax('/db/user/setUserCountryGeo', {method: 'PUT'}).then (res) ->
```

Comment:
Why use a library when we can use the [`fetch` api](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Uploading_JSON_data)?

Although it isn't fully browser supported. [We've been using it for 3 years!](https://github.com/codecombat/codecombat/blob/b0838585f7a1c022abfce308318b7cf62dc0a3f3/app/core/api/fetch-json.coffee#L27)

Reply:
No specific reason, I just used what was being used at other places in auth.coffee.
Thanks for pointing this out, will use the core/api instead which eventually uses fetch api.